src/core/Akka.Streams/ActorMaterializer.cs

```diff
@@ -86,31 +121,21 @@ public static Config DefaultConfig()
         /// <returns>TBD</returns>
         public static ActorMaterializer Create(IActorRefFactory context, ActorMaterializerSettings settings = null, string namePrefix = null)
         {
-            var haveShutDown = new AtomicBoolean();
             var system = ActorSystemOf(context);
+            
+            // forces settings to get injected into the ActorSystem the first time we materialize
+            var defaultMaterializer = DefaultMaterializer.Get(system).Materializer;
 
-            if(!InjectedConfig.TryGetValue(system, out _) && InjectedConfig.TryAdd(system, true))
-            {
-                // Inject the top-level fallback config for the Materializer once, and only once.
-                // This is a performance optimization to avoid having to do this on every materialization.
-                system.Settings.InjectTopLevelFallback(DefaultConfig());
-
-                static async Task CleanUp(ActorSystem sys)
-                {
-                    // remove ActorSystem from cache when it terminates so we don't leak memory
-                    await sys.WhenTerminated.ConfigureAwait(false);
-                    InjectedConfig.TryRemove(sys, out _);
-                    DefaultSettings.TryRemove(sys, out _);
-                }
-
-#pragma warning disable CS4014
-                CleanUp(system);
-#pragma warning restore CS4014
-
-            }
+            // optimized paths for non-allocation
+            if (context.Equals(system) && settings == null && namePrefix == null)
+                return defaultMaterializer;
+            if (context.Equals(system) && settings == null && namePrefix != null)
+                return (ActorMaterializer)defaultMaterializer.WithNamePrefix(namePrefix);
```

Comment:
Returns a new materializer instance, but avoids having to parse settings or allocate a new stream supervisor.